File: backend/src/main/java/com/bsko/ecommerce/dto/Purchase.java
package com.bsko.ecommerce.dto;

import com.bsko.ecommerce.entity.Address;
import com.bsko.ecommerce.entity.Customer;
import com.bsko.ecommerce.entity.Order;
import com.bsko.ecommerce.entity.OrderItem;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {
    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;
}

File: client/src/app/common/purchase.ts
import { Address } from "./address";
import { Customer } from "./customer";
import { Order } from "./order";
import { OrderItem } from "./order-item";

export class Purchase {
    customer: Customer;
    shippingAddress: Address;
    billingAddress;
    order: Order;
    orderItems: OrderItem[];
    
}
